### PR TITLE
Doc for the new Supervisor Mouse API

### PR DESCRIPTION
**Pull request by fabienrohrer**
ref. https://github.com/omichel/webots/pull/4930

### CODE REVIEW

**Comment by fabienrohrer** [comment]
👍

File: reference/cpp-api.md [hunk 1]
```diff
@@ -605,12 +619,13 @@ The following tables describe the C++ classes and their methods.
 | &nbsp;&nbsp; [GPS](#cpp_gps) *[getGPS](robot.md#wb_robot_get_device)(const std::string &name);                                             |
 | &nbsp;&nbsp; [Gyro](#cpp_gyro) *[getGyro](robot.md#wb_robot_get_device)(const std::string &name);                                          |
 | &nbsp;&nbsp; [InertialUnit](#cpp_inertial_unit) *[getInertialUnit](robot.md#wb_robot_get_device)(const std::string &name);                 |
-| &nbsp;&nbsp; [Joystick](#cpp_joystick) *[getJoystick](robot.md#wb_robot_get_device)();                                                     |
-| &nbsp;&nbsp; [Keyboard](#cpp_keyboard) *[getKeyboard](robot.md#wb_robot_get_device)();                                                     |
+| &nbsp;&nbsp; [Joystick](#cpp_joystick) *getJoystick();                                                                                     |
+| &nbsp;&nbsp; [Keyboard](#cpp_keyboard) *getKeyboard();                                                                                     |
```
Comment (omichel) [inline reference/cpp-api.md]: Unneeded space at the end of lines (x2).

File: reference/java-api.md [hunk 2]
```diff
@@ -634,12 +664,13 @@ The following tables describe the Java classes and their methods.
 | &nbsp;&nbsp; public [GPS](#java_gps) [getGPS](robot.md#wb_robot_get_device)(String name);                                              |
 | &nbsp;&nbsp; public [Gyro](#java_gyro) [getGyro](robot.md#wb_robot_get_device)(String name);                                           |
 | &nbsp;&nbsp; public [InertialUnit](#java_inertial_unit) [getInertialUnit](robot.md#wb_robot_get_device)(String name);                  |
-| &nbsp;&nbsp; public [Joystick](#java_joystick) [getJoystick](robot.md#wb_robot_get_device)();                                          |
-| &nbsp;&nbsp; public [Keyboard](#java_keyboard) [getKeyboard](robot.md#wb_robot_get_device)();                                          |
+| &nbsp;&nbsp; public [Joystick](#java_joystick) getJoystick();                                                                          |
+| &nbsp;&nbsp; public [Keyboard](#java_keyboard) getKeyboard();                                                                          |
```
Comment (omichel) [inline reference/java-api.md]: Extra spaces again (2x).

File: reference/mouse.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+## Mouse
+
+The [Mouse](#mouse) is a set of functions available by default for each [Robot](robot.md) node and it is used to read the computer mouse input.
```
Comment (omichel) [inline reference/mouse.md]: I'd rather say "The Mouse API provides a set of functions..."

File: reference/mouse.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+**Description**
+
+The state of the computer mouse can be read from a controller program while the simulation is running by using the functions above. 
+Firstly it is necessary to enable mouse input by calling the `wb_mouse_enable()` function.
+The `sampling_period` parameter is expressed in milliseconds, and defines how frequently readings are updated.
+Note that the first state will be available only after the first sampling period elapsed.
+After that, the state can be read by calling the `wb_mouse_get_state()` function (for more details, see [WbMouseState](#wbmousestate)).
+The measurement of the mouse state is done once at each simulation step.
```
Comment (omichel) [inline reference/mouse.md]: done &rarr; performed

Comment (omichel) [inline reference/mouse.md]: The sentence seems to contradict "The `sampling_period` parameter is expressed in milliseconds, and defines how frequently readings are updated.". It should be clarified.

**Comment by fabienrohrer** [comment]
I think the indentation issues you mentioned are correct, but difficult to interpret in the GitHub diff interface.

**Comment by fabienrohrer** [comment]
capture d ecran 2017-05-09 a 13 29 31

**Comment by fabienrohrer** [comment]
👍

File: reference/mouse.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  bool middle;
+  bool right;
+  // mouse 3D position
+  double x;
+  double y;
+  double z;
+} WbMouseState;
+```
+
+The `left`, `middle` and `right` variables are matching respectively with the left, middle and right buttons of the computer mouse.
```
Comment (omichel) [inline reference/mouse.md]: variables &rarr; fields

File: reference/mouse.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+A `true` state is matching with the pushed button, while a `false` state is matching with the released button.
```
Comment (omichel) [inline reference/mouse.md]: I'd rather say "A `true` state means the button is pressed while a `false` state means the button is released."

File: reference/mouse.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+void wb_mouse_enable(int sampling_period);
+void wb_mouse_disable();
+int wb_mouse_get_sampling_period();
+WbMouseState wb_mouse_get_state();
+```
+
+**Description**
+
+The state of the computer mouse can be read from a controller program while the simulation is running by using the functions above.
```
Comment (omichel) [inline reference/mouse.md]: &rarr; the above functions.

File: reference/mouse.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+**Description**
+
+The state of the computer mouse can be read from a controller program while the simulation is running by using the functions above. 
+Firstly it is necessary to enable mouse input by calling the `wb_mouse_enable()` function.
+The `sampling_period` parameter is expressed in milliseconds, and defines how frequently readings are updated.
+Note that the first state will be available only after the first sampling period elapsed.
+After that, the state can be read by calling the `wb_mouse_get_state()` function (for more details, see [WbMouseState](#wbmousestate)).
+The measurement of the mouse state is performed each time the `sampling_period` expires during a simulation step.
```
Comment (omichel) [inline reference/mouse.md]: Sorry but this doesn't sound clear to me... Why not simply removing this confusing sentence as the previous one is clear enough (e.g., "The `sampling_period` parameter is expressed in milliseconds, and defines how frequently readings are updated.").

File: reference/mouse.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
+## Mouse
+
+The [Mouse](#mouse) API provides a set of functions available by default for each [Robot](robot.md) node and it is used to read the computer mouse input.
```
Comment (omichel) [inline reference/mouse.md]: So, at the end it not reserved for Supervisor, but works with any robot?

Comment (fabienrohrer) [inline reference/mouse.md]: Yes, in a similar way as the Joystick and the Keyboard.

**Comment by fabienrohrer** [comment]
👍

File: reference/mouse.md [hunk 10]
```diff
@@ -24,8 +24,8 @@ typedef struct {
 } WbMouseState;
 ```
 
-The `left`, `middle` and `right` variables are matching respectively with the left, middle and right buttons of the computer mouse.
-A `true` state is matching with the pushed button, while a `false` state is matching with the released button.
+The `left`, `middle` and `right` fields are matching respectively with the left, middle and right buttons of the computer mouse.
+A `true` state means the button is pressed while a `false` state means the button is released.
 
 The `x`, `y` and `z` variables are indicating the 3D coordinate where the mouse is pointing in the 3D window.
```
Comment (omichel) [inline reference/mouse.md]: variables &rarr; fields

**Comment by fabienrohrer** [comment]
👍  (oops, sorry for this)

**Pull request merged**